tasks/files/snapshot.py

```diff
@@ -0,0 +1,813 @@
+import argparse
+import json
+import logging
+import math
+import os
+import subprocess
+import sys
+
+# from ansible.module_utils.basic import AnsibleModule
+# # TODO should move these functions to another file - but need to
+# figure out how Ansible deals with the imports
+
+logger = logging.getLogger("snapshot-role")
+
+MAX_LVM_NAME = 127
+
+
+class LvmBug(RuntimeError):
+    """
+    Things that are clearly a bug with lvm itself.
+    """
+
+    def __init__(self, msg):
+        super().__init__(msg)
+
+    def __str__(self):
+        return "lvm bug encountered: %s" % " ".join(self.args)
+
+
+class SnapshotStatus:
+    SNAPSHOT_OK = 0
+    ERROR_INSUFFICENT_SPACE = 1
+    ERROR_ALREADY_DONE = 2
+    ERROR_SNAPSHOT_FAILED = 3
+    ERROR_REMOVE_FAILED = 4
+    ERROR_REMOVE_FAILED_NOT_SNAPSHOT = 5
+    ERROR_LVS_FAILED = 6
+    ERROR_NAME_TOO_LONG = 7
+    ERROR_ALREADY_EXISTS = 8
+    ERROR_NAME_CONFLICT = 9
+    ERROR_VG_NOTFOUND = 10
+    ERROR_LV_NOTFOUND = 11
+    ERROR_VERIFY_NOTSNAPSHOT = 12
+    ERROR_VERIFY_COMMAND_FAILED = 13
+    ERROR_VERIFY_NOT_FOUND = 14,
+    ERROR_CMD_INVALID = 15
+    ERROR_VERIFY_REMOVE_FAILED = 16,
+    ERROR_VERIFY_REMOVE_SOURCE_SNAPSHOT = 17
+
+
+
+# what percentage is part of whole
+def percentage(part, whole):
+    return 100 * float(part) / float(whole)
+
+
+# what is number is percent of whole
+def percentof(percent, whole):
+    return float(whole) / 100 * float(percent)
+
+
+def set_up_logging(log_dir="/tmp", log_prefix="snapshot_role"):
+    logger.setLevel(logging.DEBUG)
+
+    def make_handler(path, prefix, level):
+        log_file = "%s/%s.log" % (path, prefix)
+        log_file = os.path.realpath(log_file)
+        handler = logging.FileHandler(log_file)
+        handler.setLevel(level)
+        formatter = logging.Formatter(
+            "%(asctime)s %(levelname)s %(name)s/%(threadName)s: %(message)s"
+        )
+        handler.setFormatter(formatter)
+        return handler
+
+    handler = make_handler(log_dir, log_prefix, logging.DEBUG)
+    stdout_handler = logging.StreamHandler(stream=sys.stdout)
+
+    logger.addHandler(handler)
+    logger.addHandler(stdout_handler)
```

Comment:
```suggestion
    # logger.addHandler(stdout_handler)
```
There is a lot of output for the user.  I think the stdout handler is only needed for debugging.  We could figure out a way for the user to control the verbosity by using `ansible-playbook -vvv...` in order to see the full output.  But for now it's easier to view in the logs and parse the output.

Reply:
ok